packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -1407,10 +1407,42 @@ class ResizeImage extends ImageProvider<ResizeImageKey> {
   }
 }
 
+/// The strategy for [Image.network] and [NetworkImage] to decide whether to
+/// display images in <img> elements in a platform view instead of fetching
+/// bytes.
+///
+/// See [Image.network] for more explanation on the impact.
+///
+/// This option is only effective on the Web platform. Other platforms always
+/// display network images by fetching bytes.
```

Comment:
Somewhere in the docs we should link to https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy, which is the main driver for all this.